go-controller/pkg/ovn/pods.go

```diff
@@ -327,8 +344,14 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	}
 
 	if annotation == nil {
-		gwIfAddr := util.GetNodeGatewayIfAddr(nodeSubnet)
-		routes, gwIP, err := getRoutesGatewayIP(pod, gwIfAddr)
+		hybridOverlayExternalGW := &net.IP{}
+		if config.HybridOverlay.Enabled {
+			hybridOverlayExternalGW, err = oc.getHybridOverlayExternalGwAnnotation(pod.Namespace)
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
Providing we now have two annotations, shouldn't we check for VTEP annotation here as well?

Reply:
what would the check do? print a warning or throw an error? I don't think we want to throw an error and stop pods from coming up do we?